### PR TITLE
feat(web): user profile bar and logout

### LINKED ISSUE #225: Logout button
### Check for existing issues

- [X] Completed

### Description

A login button exists, but no possibility to logout. It would be cool to be able to end a session from within the website.

### Do you want to help us with the new feature? (If "Yes", we assign you the ticket)

No

### PR DESCRIPTION
**Pull request by rawkode**
Closes #225

### CODE REVIEW

File: projects/rawkode-academy/web/src/components/auth/profile.vue [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<script setup lang="ts">
+import { initFlowbite } from 'flowbite';
+import { onMounted } from 'vue';
+
+onMounted(() => {
+	initFlowbite();
```
Comment (icepuma) [inline projects/rawkode-academy/web/src/components/auth/profile.vue:6]: Hmm, is https://github.com/RawkodeAcademy/RawkodeAcademy/blob/main/projects/rawkode-academy/web/src/wrappers/page.astro#L21 not enough or is Vue special here and needs a second round of init?

File: projects/rawkode-academy/web/src/middleware.ts [hunk 2]
```diff
@@ -27,49 +28,55 @@ export const onRequest = defineMiddleware(async (context, next) => {
...
+					clientId,
+					refreshToken: session.refreshToken,
+				});
+
+			const encryptedSession = await sealData(
+				{
+					accessToken,
+					refreshToken,
+				},
+				{ password: getSecret("WORKOS_COOKIE_PASSWORD") || "" },
```
Comment (icepuma) [inline projects/rawkode-academy/web/src/middleware.ts:60]: Is `""` a sane default?

File: projects/rawkode-academy/web/src/pages/auth/logout.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import WorkOS from "@workos-inc/node";
+import type { APIRoute } from "astro";
+import { decodeJwt } from "jose";
+import { getSessionFromCookie } from "../../middleware";
+
```
Comment (icepuma) [inline projects/rawkode-academy/web/src/pages/auth/logout.ts:5]:
```typescript
import { WORKOS_API_KEY } from "astro:env/server";
```

should work as well :)

File: projects/rawkode-academy/web/src/components/auth/profile.vue (hunk 1 above)
Comment (icepuma) [reply projects/rawkode-academy/web/src/components/auth/profile.vue:6]: Ah nevermind, saw the removal of said piece of code

File: projects/rawkode-academy/web/src/pages/auth/logout.ts (hunk 3 above)
Comment (rawkode) [reply projects/rawkode-academy/web/src/pages/auth/logout.ts:5]: I prefer the getSecret approach 😀

File: projects/rawkode-academy/web/src/middleware.ts (hunk 2 above)
Comment (rawkode) [reply projects/rawkode-academy/web/src/middleware.ts:60]: No, we should probably error out

**Review by icepuma — DISMISSED** [review]

**Comment by rawkode** [comment]
Can't work out what is wrong with that final test :(

**Pull request merged**